machinery-helper/machinery_helper_test.go

```diff
@@ -0,0 +1,191 @@
+}
+
+func TestAddImplicitlyManagedDirs(t *testing.T) {
+  files_original := map[string]string{
+    "/abc/def/ghf/somefile": "",
+    "/zzz": "/abc/def",
+  }
+  dirs_original := map[string]bool{
+    "/abc/def": true,
+  }
+  dirs_expected := map[string]bool{
+    "/abc": false,
+    "/abc/def": true,
+    "/abc/def/ghf": false,
+    "/zzz": false,
+  }
+
+  dirs := dirs_original
+
+  addImplicitlyManagedDirs(dirs, files_original)
+
+  if !reflect.DeepEqual(dirs, dirs_expected) {
+    t.Errorf("addImplicitlyManagedDirs('%v') = '%v', want '%v'", dirs_original, dirs, dirs_expected)
+  }
+}
+
+func TestAssembleJson(t *testing.T) {
+  unmanaged_files_map := map[string]string{
+    "name": "/usr/share/go_rulez", "type":"file",
+  }
+  want := `{
+   "extracted": false,
+   "files": {
+     "name": "/usr/share/go_rulez",
+     "type": "file"
+   }
+ }`
+
+  json := assembleJson(unmanaged_files_map)
+  if !reflect.DeepEqual(json, want) {
+    t.Errorf("assembleJson() = '%v', want '%v'", json, want)
+  }
+}
+
+func TestRespectManagedDirsInUnmanagedDirs(t *testing.T) {
+  unmanaged_files := make(map[string]string)
+  want_unmanaged_files := make(map[string]string)
+
+  /*
+  We mock the readDir method and return following directory structure:
+   /
+   /managed_dir/
+   /managed_dir/unmanaged_dir/
+   /managed_dir/unmanaged_dir/managed_dir/
+  */
+  readDir = func(dir string) ([]os.FileInfo, error) {
+    dirs := make([]os.FileInfo, 0, 1)
+    var fi os.FileInfo
+    switch dir {
+    case "/":
+      fi = fakefileinfo.New("managed_dir", int64(123), os.ModeType, time.Now(), true, nil)
+    case "/managed_dir/":
+      fi = fakefileinfo.New("unmanaged_dir", int64(123), os.ModeType, time.Now(), true, nil)
+    case "/managed_dir/unmanaged_dir/":
+      fi = fakefileinfo.New("managed_dir", int64(123), os.ModeType, time.Now(), true, nil)
+    }
+    dirs = append (dirs, fi)
+    return dirs, nil
+  }
+
+  rpm_files := make(map[string]string)
+
+  rpm_dirs := map[string]bool{
+    "/managed_dir": true,
+    "/managed_dir/unmanaged_dir/managed_dir": true,
+  }
+
+  findUnmanagedFiles("/", rpm_files, rpm_dirs, unmanaged_files)
+
+  if !reflect.DeepEqual(unmanaged_files, want_unmanaged_files) {
+    t.Errorf("findUnmanagedFiles() = '%v', want '%v'", unmanaged_files, want_unmanaged_files)
+  }
+}
+
+func TestHasManagedDirs(t *testing.T) {
+  rpm_dirs := map[string]bool{
+    "/managed_dir": true,
+    "/managed_dir/unmanaged_dir/sub_dir/managed_dir": true,
+  }
+
+  has_managed_dirs := hasManagedDirs("/managed_dir/unmanaged_dir", rpm_dirs)
+  want := true
+
+  if has_managed_dirs != want {
+    t.Errorf("hasManagedDirs() = '%v', want '%v'", has_managed_dirs, want)
+  }
+
+}
+
+func TestSubdirIsNotAccidentallyConsideredManaged(t *testing.T) {
+  rpm_dirs := map[string]bool{
+    "/usr": true,
+    "/usr/foobar": true,
+  }
+
+  has_managed_dirs := hasManagedDirs("/usr/foo", rpm_dirs)
```

Comment:
don't use underscores in Go names; var has_managed_dirs should be hasManagedDirs